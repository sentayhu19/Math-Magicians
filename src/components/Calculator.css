.calc-wrap {
  position: fixed;
  right: 250px;
  top: 220px;
}
