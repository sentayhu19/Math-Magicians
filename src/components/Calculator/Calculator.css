.calc-wrap {
  position: fixed;
  right: 250px;
  top: 220px;
}

@media only screen and (max-width: 1024px) {
  .calc-wrap {
    position: static;
    margin: auto;
  }
}
