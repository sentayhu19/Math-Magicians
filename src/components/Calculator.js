import React, { useState } from 'react';
import calculate from '../logic/calculate';
import './Calculator.css';

const Calculator = () => {
  const [currentCalc, setcalc] = useState({
    next: '',
    total: 0,
    operation: '',
  });
  const { next, total, operation } = currentCalc;
  const handleClick = (e) => {
    e.preventDefault();
    setcalc((state) => calculate({
      next: state.next,
      total: state.total,
      operation: state.operation,
    },
    e.target.innerHTML));
  };

  return (
    <>
      <h2>Lets do some math!</h2>
      <div className="calc-wrap">
        <div id="input">
          {total}
          {' '}
          {operation}
          {' '}
          {next}
        </div>
        <div id="row1">
          <button type="button" className="op" onClick={handleClick}>AC</button>
          <button type="button" className="op" onClick={handleClick}>+/-</button>
          <button type="button" className="op" onClick={handleClick}>%</button>
          <button type="button" className="operator" onClick={handleClick}>รท</button>
        </div>
        <div id="row2">
          <button type="button" className="number" id="7" onClick={handleClick}>7</button>
          <button type="button" className="number" id="8" onClick={handleClick}>8</button>
          <button type="button" className="number" id="9" onClick={handleClick}>9</button>
          <button type="button" className="operator" onClick={handleClick}>x</button>
        </div>
        <div id="row3">
          <button type="button" className="number" id="4" onClick={handleClick}>4</button>
          <button type="button" className="number" id="5" onClick={handleClick}>5</button>
          <button type="button" className="number" id="6" onClick={handleClick}>6</button>
          <button type="button" className="operator" onClick={handleClick}>-</button>
        </div>

        <div id="row4">
          <button type="button" className="number" id="1" onClick={handleClick}>1</button>
          <button type="button" className="number" id="2" onClick={handleClick}>2</button>
          <button type="button" className="number" id="3" onClick={handleClick}>3</button>
          <button type="button" className="operator" onClick={handleClick}>+</button>
        </div>
        <div id="row5">
          <button type="button" id="n0" onClick={handleClick}>0</button>
          <button type="button" className="number" onClick={handleClick}>.</button>
          <button className="operator" type="button" onClick={handleClick}>=</button>
        </div>
      </div>
    </>
  );
};
export default Calculator;
