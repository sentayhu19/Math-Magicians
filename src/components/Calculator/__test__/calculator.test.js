/* eslint-disable import/extensions */
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Calculator page of the App', () => {
  test('Builds the snapchot of the Calculator component safely', () => {
    const rend = renderer.create(<Calculator />);
    expect(rend.toJSON()).toMatchSnapshot();
  });
  test('Render Calculator component', () => {
    render(<Calculator />);
  });
});
