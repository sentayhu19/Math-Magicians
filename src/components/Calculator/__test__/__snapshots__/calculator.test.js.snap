// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page of the App Builds the snapchot of the Calculator component safely 1`] = `
Array [
  <h2>
    Lets do some math!
  </h2>,
  <div
    className="calc-wrap"
  >
    <div
      id="input"
    >
      0
       
       
    </div>
    <div
      id="row1"
    >
      <button
        className="op"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="op"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="op"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      id="row2"
    >
      <button
        className="number"
        id="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="number"
        id="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="number"
        id="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      id="row3"
    >
      <button
        className="number"
        id="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="number"
        id="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="number"
        id="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      id="row4"
    >
      <button
        className="number"
        id="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="number"
        id="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="number"
        id="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      id="row5"
    >
      <button
        id="n0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
